### ISSUE TITLE
Windows 7 Enterprise feedback on pharo-launcher-bleedingEdge-signed.msi 2018-04-16 16:03

### ISSUE THREAD

**Issue by rjsargent** [pharo-project/pharo-launcher#93]
[It would be nice if one could copy the text from the renders page for http://files.pharo.org/pharo-launcher/win-alpha/]

1. I downloaded and installed the launcher in APPDATA. When I attempted to run it, the executable was  blocked by group policy. (I haven't seen a group policy setting interfere with me for at least four years,.)
![1-blockedbygrouppolicy](https://user-images.githubusercontent.com/4140292/38823289-d792d206-415a-11e8-93c3-5f2b734a17ad.png)

2. I tried to run it as administrator and that worked. However, it reported an unknown published. (The MSI file was signed, but not the pharo.exe.)
![2-unknownpublisher](https://user-images.githubusercontent.com/4140292/38823362-15c3bcac-415b-11e8-94f0-13f392849387.png)

3. This point seems to have been reported already in its own issue. I'm including it just for completeness. I selected the Pharo image (stable) from the Downloaded templates and received a walkback.
![3-walkback](https://user-images.githubusercontent.com/4140292/38823489-55ae1f88-415b-11e8-8789-83bda9c4c990.png)

4. When I expanded the RMoD Jenkins folder, I received a network error. I could not discover any way to retry this. Of course, it might have been automatic, in which case the problem is that there was inadequate feedback.
![4-networkerror](https://user-images.githubusercontent.com/4140292/38823613-b3661f4a-415b-11e8-98ae-cdea23669ba7.png)

5. I could not find anything in the launcher UI to identify the version of the launcher. Some kind of "About" facility from which one could copy (or at least transcribe) the build identification would be a good idea.

----------------------
I will also attempt this exercise against my Windows 10 home computer.

**Comment by demarey** [comment]
> [It would be nice if one could copy the text from the renders page for http://files.pharo.org/pharo-launcher/win-alpha/]

I took a look. We use [h5ai](https://larsjung.de/h5ai/). I do not see any option to enable copy of the text.

> 1. I downloaded and installed the launcher in APPDATA. When I attempted to run it, the executable was  blocked by group policy. (I haven't seen a group policy setting interfere with me for at least four years,.)
![1-blockedbygrouppolicy](https://user-images.githubusercontent.com/4140292/38823289-d792d206-415a-11e8-93c3-5f2b734a17ad.png)

Do you mean APPDATA or LOCALAPPDATA? If it is APPDATA, then I think it is expected that you got a group policy warning: Application Data can be redirected through Group Policies to a network location. I think Local Application Data should not be affected by the group policy (but no way to test it).

> 2. I tried to run it as administrator and that worked. However, it reported an unknown published. (The MSI file was signed, but not the pharo.exe.)
![2-unknownpublisher](https://user-images.githubusercontent.com/4140292/38823362-15c3bcac-415b-11e8-94f0-13f392849387.png)

Yes, you're right. I did not get this warning when testing but, yes, pharo.exe should also be signed.

> 3. This point seems to have been reported already in its own issue. I'm including it just for completeness. I selected the Pharo image (stable) from the Downloaded templates and received a walkback.
![3-walkback](https://user-images.githubusercontent.com/4140292/38823489-55ae1f88-415b-11e8-8789-83bda9c4c990.png)

yes: #41 

> 4. When I expanded the RMoD Jenkins folder, I received a network error. I could not discover any way to retry this. Of course, it might have been automatic, in which case the problem is that there was inadequate feedback.
![4-networkerror](https://user-images.githubusercontent.com/4140292/38823613-b3661f4a-415b-11e8-98ae-cdea23669ba7.png)

I should remove this one. There is a network error because it is a private Jenkins requiring authentication.

> 5. I could not find anything in the launcher UI to identify the version of the launcher. Some kind of "About" facility from which one could copy (or at least transcribe) the build identification would be a good idea.

There is one ;) You have an "About" box available with the question mark button at the bottom-right of the UI

Thanks for this detailed and useful report!

**Comment by rjsargent** [comment]
Thanks for the answers and clarification.
(I still "owe" a report from a yet-to-be-done attempt on my Windows 10 system at home.)


1. APPDATA vs LOCALAPPDATA: you are correct. It is in the AppData/Local sub-directory.

However, I noticed that the AppData\Roaming had a new directory named pharo created 7 minutes earlier than the successful one for the Pharo Launcher. This appears to be the right amount of time for the first attempt (which was block by the group policy).

The Roaming/pharo directory contains exactly one directory, templateCache, which in turn contains exactly one file, cache.json. This file is empty.


5. The "About" information could use some work. :-)
Pharo Launcher version: N/A
Packages :
- PharoLauncher-Core-VincentBlondeau.184
- PharoLauncher-Spec-ChristopheDemarey.66
- PharoLauncher-Tests-Core-VincentBlondeau.24

**Comment by demarey** [comment]
> 1. APPDATA vs LOCALAPPDATA: you are correct. It is in the AppData/Local sub-directory.
>
> However, I noticed that the AppData\Roaming had a new directory named pharo created 7 minutes earlier than the successful one for the Pharo Launcher. This appears to be the right amount of time for the first attempt (which was block by the group policy).

So, you mean that Pharo Launcher first tried to install itself in APPDATA?
It is strange because the default installation folder is set to LOCALAPPDATA. Could you check or do you remember what was the default folder displayed in the installer wizard?

In addition, does it work if you install PharoLauncher in LOCALAPPDATA?

> 5. The "About" information could use some work. :-)
> Pharo Launcher version: N/A
> Packages :
> - PharoLauncher-Core-VincentBlondeau.184
> - PharoLauncher-Spec-ChristopheDemarey.66
> - PharoLauncher-Tests-Core-VincentBlondeau.24

Yes, it is not very nice because this version is a bleeding edge version. There is no released version found. For a released version you should see: `Pharo Launcher version: 1.2`
